packages/ui-library/src/components/tooltip-container/index.js

```diff
@@ -74,15 +86,48 @@ TooltipContainer.propTypes = {
  * @returns {JSX.Element} The element.
  */
 export const TooltipTrigger = ( { as: Component = "button", className, children, ariaDescribedby, ...props } ) => {
-	const { show } = useTooltipContext();
+	const { show, hide, tooltipPosition, isVisible } = useTooltipContext();
+	const triggerRef = useRef();
+
+	useEffect( () => {
+		const handleMouseMove = debounce( ( event ) => {
+			const rect = triggerRef.current.getBoundingClientRect();
+			const extendedRect = {
+				top: rect.top - GRACE_MRGIN,
+				right: rect.right + GRACE_MRGIN,
+				bottom: rect.bottom + GRACE_MRGIN,
+				left: rect.left - GRACE_MRGIN,
+			};
+
+			const mouseX = event.clientX;
+			const mouseY = event.clientY;
+			const outsideTooltip = mouseX < tooltipPosition.left ||
+				mouseX > tooltipPosition.right || mouseY < tooltipPosition.top || mouseY > tooltipPosition.bottom;
+			const outsideTriggerWithMargin = mouseX < extendedRect.left ||
+				mouseX > extendedRect.right || mouseY < extendedRect.top || mouseY > extendedRect.bottom;
+
+			if ( outsideTriggerWithMargin && document.activeElement !== triggerRef.current && outsideTooltip ) {
+				hide();
+			} else {
+				show();
+			}
+		}, DEBOUNCE_DELAY );
+
+		document.addEventListener( "pointermove", handleMouseMove );
+
+		return () => {
+			document.removeEventListener( "pointermove", handleMouseMove );
+			handleMouseMove.cancel();
+		};
+	}, [ show, hide, triggerRef, tooltipPosition, isVisible ] );
```

Comment:
I don't think `triggerRef` every changes. Do you mean `triggerRef.current` to verify the DOM node / memory address?